### PR TITLE
Refactor!: generator now always copies, making transforms much simpler

### PR DESCRIPTION
**Pull request by tobymao**

### CODE REVIEW

**Comment by tobymao** [comment]
this is a v19 change

File: sqlglot/optimizer/simplify.py [hunk 1]
```diff
@@ -1070,3 +1067,69 @@ def _flat_simplify(expression, simplifier, root=True):
                 lambda a, b: expression.__class__(this=a, expression=b), operands
             )
     return expression
+
+
+def gen(expression: t.Any) -> str:
+    """Simple pseudo sql generator for quickly generating sortable and uniq strings.
+
+    Sorting and deduping sql is a necessary step for optimization. Calling the actual
+    generator is expensive so we have a bare minimum sql generator here.
```
Comment (GeorgeSittas) [inline sqlglot/optimizer/simplify.py:1076]: why is calling the generator expensive? can't we set `copy=False` or something? this seems do add quite some complexity

**Comment by GeorgeSittas** [comment]
Haven't checked yet thoroughly but I'd be a bit cautious with the copies in `transform.py` because the assumption before was that each expression would be copied by default so maybe some copies were intentional? Just thinking out loud

File: sqlglot/optimizer/simplify.py (hunk 1 above)
Comment (tobymao) [reply sqlglot/optimizer/simplify.py:1076]: generator is super expensive, so i needed to add a cache

Comment (GeorgeSittas) [reply sqlglot/optimizer/simplify.py:1076]: we can't bring it back?

Comment (tobymao) [reply sqlglot/optimizer/simplify.py:1076]: it’s not necessary anymore and removes complexity

File: sqlglot/generator.py [hunk 2]
```diff
@@ -2962,15 +2948,6 @@ def _simplify_unless_literal(self, expression: E) -> E:
         if not isinstance(expression, exp.Literal):
             from sqlglot.optimizer.simplify import simplify
 
-            expression = simplify(expression.copy())
+            expression = simplify(expression)
 
         return expression
-
-
-def cached_generator(
```
[the remark this answers is not in the document]
Comment (tobymao) [reply sqlglot/generator.py:2970]: correct, needed the cache because sql gen was so expensive

**Pull request merged**